### PR TITLE
cmd/atlasci: add integrity checks to summary report

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: sql/migrate/migrate.go [hunk 1]
```diff
@@ -980,15 +980,12 @@ var (
 // If they don't match ErrChecksumMismatch is returned.
 func Validate(dir Dir) error {
 	fh, err := readHashFile(dir)
-	if os.IsNotExist(err) {
+	if errors.Is(err, fs.ErrNotExist) {
```
Comment (masseelch) [inline sql/migrate/migrate.go:983]: 🙏

File: sql/migrate/migrate.go [hunk 2]
```diff
@@ -980,15 +980,12 @@ var (
...
 		// If there are no migration files yet this is okay.
 		files, err := fs.ReadDir(dir, "/")
-		if err != nil {
+		if err == nil && len(files) == 0 {
 			return err
 		}
-		if len(files) == 0 {
-			return nil
-		}
```
Comment (masseelch) [inline sql/migrate/migrate.go:991]: I prefer to explicitly return nil instead of a variable.

File: sql/mysql/driver_test.go [hunk 3]
```diff
@@ -68,12 +68,12 @@ func TestDriver_LockError(t *testing.T) {
...
 
 	t.Run("Timeout", func(t *testing.T) {
-		name, sec := "name", 1
+		name, sec := "name", 60
 		m.ExpectQuery(sqltest.Escape("SELECT GET_LOCK(?, ?)")).
 			WithArgs(name, sec).
 			WillReturnRows(sqlmock.NewRows([]string{"acquired"}).AddRow(0)).
 			RowsWillBeClosed()
-		unlock, err := d.Lock(context.Background(), name, time.Second)
+		unlock, err := d.Lock(context.Background(), name, time.Minute)
```
Comment (zeevmoney) [inline sql/mysql/driver_test.go:76]: why did you change this?

File: cmd/atlasci/internal/ci/run_test.go [hunk 4]
```diff
@@ -56,22 +56,6 @@ func TestRunner_Run(t *testing.T) {
 `, b.String())
 }
 
-//go:embed testdata/atlas.sum
-var hash []byte
-
-func TestChecksumAnalyzer_Analyze(t *testing.T) {
```
Comment (zeevmoney) [inline cmd/atlasci/internal/ci/run_test.go:62]: Why not test this?

**Review by zeevmoney — APPROVED** [review]
see questions

File: sql/mysql/driver_test.go (hunk 3 above)
Comment (a8m) [reply sql/mysql/driver_test.go:76]: It was failing in the ci due to "context canceled".

**Review by masseelch — APPROVED** [review]
